src/guides/v2.4/comp-mgr/dev_add-update.md

```diff
@@ -0,0 +1,79 @@
+---
+group: software-update-guide
+title: Add or update components
+menu_title: Add or update components
+menu_order: 5
+menu_node:
+functional_areas:
+  - Upgrade
+redirect_from: /guides/v.24/install-gde/install/cli/dev_add-update.html
+---
+
+A contributing developer updates components by specifying components and their versions in Magento's `composer.json`.
+
+To update components if you're *not* a contributing developer, see [Updating the Magento application and components]({{ page.baseurl }}/comp-mgr/bk-compman-upgrade-guide.html).
+
+You can either add a `require` section to `composer.json` or you can use the `composer require` command as follows:
+
+1. Log in to the Magento server, or switch to, the [Magento file system owner](https://glossary.magento.com/magento-file-system-owner).
+1. Change to the directory to which you cloned the Magento application. For example,
+
+   ```bash
+   cd /var/www/magento2
+   ```
+
+You have the following options:
+
+### Get available module versions
+
+Command usage:
+
+```bash
+composer show --all <vendor>/<name>
+```
+
+For example,
+
+```bash
+composer show --all example/module
```

Comment:
Is this a working example? If not, can it be?

Reply:
I don't know. Maybe you can try it out!